coderd/inboxnotifications.go

```diff
@@ -28,6 +29,45 @@ const (
 	notificationFormatPlaintext = "plaintext"
 )
 
+const (
+	FallbackIconWorkspace = "DEFAULT_WORKSPACE_ICON"
+	FallbackIconAccount   = "DEFAULT_ACCOUNT_ICON"
+	FallbackIconTemplate  = "DEFAULT_TEMPLATE_ICON"
+	FallbackIconOther     = "DEFAULT_OTHER_ICON"
+)
+
+var fallbackIcons = map[uuid.UUID]string{
+	// workspace related notifications
+	notifications.TemplateWorkspaceCreated:           FallbackIconWorkspace,
+	notifications.TemplateWorkspaceCreated:           FallbackIconWorkspace,
+	notifications.TemplateWorkspaceManuallyUpdated:   FallbackIconWorkspace,
+	notifications.TemplateWorkspaceDeleted:           FallbackIconWorkspace,
+	notifications.TemplateWorkspaceAutobuildFailed:   FallbackIconWorkspace,
+	notifications.TemplateWorkspaceDormant:           FallbackIconWorkspace,
+	notifications.TemplateWorkspaceAutoUpdated:       FallbackIconWorkspace,
+	notifications.TemplateWorkspaceMarkedForDeletion: FallbackIconWorkspace,
+	notifications.TemplateWorkspaceManualBuildFailed: FallbackIconWorkspace,
+	notifications.TemplateWorkspaceOutOfMemory:       FallbackIconWorkspace,
+	notifications.TemplateWorkspaceOutOfDisk:         FallbackIconWorkspace,
+
+	// account related notifications
+	notifications.TemplateUserAccountCreated:           FallbackIconAccount,
+	notifications.TemplateUserAccountDeleted:           FallbackIconAccount,
+	notifications.TemplateUserAccountSuspended:         FallbackIconAccount,
+	notifications.TemplateUserAccountActivated:         FallbackIconAccount,
+	notifications.TemplateYourAccountSuspended:         FallbackIconAccount,
+	notifications.TemplateYourAccountActivated:         FallbackIconAccount,
+	notifications.TemplateUserRequestedOneTimePasscode: FallbackIconAccount,
+
+	// template related notifications
+	notifications.TemplateTemplateDeleted:             FallbackIconTemplate,
+	notifications.TemplateTemplateDeprecated:          FallbackIconTemplate,
+	notifications.TemplateWorkspaceBuildsFailedReport: FallbackIconTemplate,
+
+	// other related notifications
+	notifications.TemplateTestNotification: FallbackIconOther,
+}
```

Comment:
Can you add a test to ensure that each notification template type has a fallback icon associated with it?

Reply:
Added some tests 👀